### DISCUSSION TITLE
How to restrict access to a relation only via nested object?

### DISCUSSION THREAD

**Discussion by murtuza-ranapur** [hasura/graphql-engine#8477] [Question] [answered]
For example, I have 2 tables `users` and `additional_users_data` who have a one-to-one relationship between them. What I want is for my users to access the `additional_users_data` only via the `users` table and not directly. Is there a way to achieve this? 

Some additional info:
1. We are tracking our existing database tables and relationships.
2. We are using Hasura's role-based authorization to give access to columns and remote schema queries.

Possible alternatives that I can think of (But would like to avoid if a solution is available)
1. Creating a view and give users access to that
2. Merge `additional_users_data` to `users` table

I tried revoking permission of the PK field of the `additional_users_data` table but only removed the `additional_users_data_by_pk` query it is still generating the `additional_users_data` query.

**Comment by renato** [comment, marked as answer]
It looks like it's been handled on the latest beta: https://github.com/hasura/graphql-engine/releases/tag/v2.8.0-beta.1

**Reply by seanparkross** [reply]
Yes, brand new feature to disable root fields incoming 🪂
